common/static/common/js/discussion/views/discussion_content_view.js

```diff
@@ -493,8 +493,16 @@
             };
 
             DiscussionContentShowView.prototype.getAuthorDisplay = function() {
+                var firstName, lastName, displayName;
+                firstName = this.model.get('first_name');
+                lastName = this.model.get('last_name');
+                if (typeof firstName === 'undefined' || typeof lastName === 'undefined') {
+                    displayName = this.model.get('username');
+                } else {
+                    displayName = firstName + ' ' + lastName[0];
+                }
```

Comment:
Looks good just a few queries. Is the condition tested? Plus shouldn't we create a method for getting the name of user ? A suggestion, i think we better use ternary operator here.

Reply:
@hamzamunir7300 I have addressed your comment.